ImGuiApp/ImGuiController/ImGuiController.cs

```diff
@@ -427,6 +456,61 @@ private static ImGuiKey TranslateInputKeyToImGuiKey(Key key)
 		};
 	}
 
+	private static bool TryGetImGuiMouseButton(MouseButton button, out ImGuiMouseButton imguiMouseButton)
+	{
+		if (button is MouseButton.Left)
+		{
+			imguiMouseButton = ImGuiMouseButton.Left;
+			return true;
+		}
+		else if (button is MouseButton.Right)
+		{
+			imguiMouseButton = ImGuiMouseButton.Right;
+			return true;
+		}
+		else if (button is MouseButton.Middle)
+		{
+			imguiMouseButton = ImGuiMouseButton.Middle;
+			return true;
+		}
+
+		imguiMouseButton = ImGuiMouseButton.COUNT;
+		return false;
+	}
+
+	/// <summary>
+	/// Tries to get the matching mod key for an ImGuiKey.
+	/// </summary>
+	/// <param name="key">The ImGuiKey to convert.</param>
+	/// <param name="modKey">The matching mod key.</param>
+	/// <returns>If there is an associated mod key.</returns>
+	private static bool TryGetImGuiModKey(ImGuiKey key, out ImGuiKey modKey)
```

Comment:
Rename to `TranslateImGuiModKey` and convert to a switch expression throwing on invalid values.

You can use a warning suppression attribute on this method to ignore the error about not handling all values, so that you dont need to list every possible key. It will be a suggested quick fix on the error.

Reply:
Also does not need the TryGet pattern here.

Reply:
I don't understand how we can throw on invalid values and not implement all case statements. 

I'll do a first pass change of making it a switch expression and returning COUNT on invalid keys. Seems "better" than adding the warning ignore?

Reply:
Implemented changes as I've described above.